chichiwang/exercise-js
# Create a default page layout

## Description
Linked to Issue: #75 
Create a default layout for pages `pages/Layout` to wrap around application pages. The default layout will contain a navigation bar `pages/Layout/Nav.tsx` which links to all pages in the application, displays the project logo, and links to [this Github project](https://github.com/chichiwang/tamsui).

## Changes
* Update `.eslintrc` to disable rule [react/prop-types](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/prop-types.md) for `.js` files
  * Specifically so it is not required for mock components in test files (which all contain the `.js` extension)
* Update global style rules in `app/global.module.scss`
  * Give base font rules to the `body` element
  * Set base box-sizing rule to all elements
  * Re-style links
* Add scss variables in `app/var.module.scss`
  * Add site colors
  * Add content layout values
* Add dependency `classnames` to `package.json`
* Add script `prewatch` to `package.json` to clean and build the `/dist` directory
* Create a default layout component `pages/Layout/index.tsx`
* Create a default navigation bar component `pages/Layout/Nav.tsx`
  * Render the nav in the default layout
* Add [Github logo](https://github.com/logos) for use in the nav bar to link to this project repo
* Wrap the Home component `pages/Home/index.tsx` in the default Layout component
* Wrap the Counter component `pages/Counter/index.tsx` in the default Layout component
* Wrap the NotFound component `pages/NotFound/index.tsx` in the default Layout component
* Wrap the ErrorPage component `pages/ErrorPage/index.tsx` in the default Layout component

## Steps to QA
* Pull down and switch to this branch
* Run the server with `npm run dev`/`npm run prod`/`npm run watch` and verify the page looks good in the browser
* Check the links in the nav bar to ensure they work as intended
  * The Github social logo should link to this project repo in a new tab
* Verify that the `NotFound` page component renders with the layout/nav (navigate to a non-existent route)
* Verify the `ErrorPage` page component renders the layout/nav
  * Throw an error from a page component, to avoid errors you can use:
```javascript
if (Math.random() >= 0) {
  throw new Error('Oops!');
}
```

## Not Done in This PR (will complete in a follow-up)
* Print styles for the Layout/Nav
* Responsive styles Layout/Nav
